docs/development/linux-bridge-primary-nic-scenario.md

```diff
@@ -0,0 +1,787 @@
+# NMPolicy stages with bridge at primary NIC scenario
+
+The NMPolicy processing works like a pipeline where the chain like this:
+
+1. Calculate captured state
+   1. Lexer: convert capture expression to tokens
+   2. Parser: convert capture tokens to AST
+   3. Emitter: convert AST + current state into captured state 
+2. Calculate desired state
```

Comment:
Why do we need the lexer->parser algorithm to resolve the desired state place holders?
Those holders are just a path to the captures yaml. It should very simple to parse the placeholders without the need to tokenize + ast it.
The first part of the place holder path is the key name of the capture. The rest is the path in the captured snippet ([gjson](https://github.com/tidwall/gjson) can get used to retrieve the value from the snippet `gjson.Get(capturedSnippet, path)`)

Reply:
Let's not introduce another lib, we have already a tool to do the job and is good enough for our case the fact that it uses an ast is an implementation detail.